packages/crypto/src/bls12_318/aggregate.rs

```diff
@@ -0,0 +1,229 @@
+use crate::{errors::InvalidPoint, CryptoError};
+
+use super::points::{g1_from_fixed, g2_from_fixed, G1, G2};
+
+const G1_POINT_SIZE: usize = 48;
+const G2_POINT_SIZE: usize = 96;
+
+/// Takes a list of points in G1 (48 bytes each) and aggregates them.
+///
+/// This is like Aggregate from <https://datatracker.ietf.org/doc/html/draft-irtf-cfrg-bls-signature-05>
+/// but works for signatures as well as public keys.
+pub fn bls12_381_aggregate_g1(points: &[u8]) -> Result<[u8; 48], CryptoError> {
+    if points.len() % G1_POINT_SIZE != 0 {
+        return Err(InvalidPoint::DecodingError {}.into());
```

Comment:
Here it would be great to have a specific error type for the cases
- empty input
- number of bytes cannot be devided by point size